packages/flutter_test/lib/src/test_pointer.dart

```diff
@@ -15,13 +15,34 @@ export 'dart:ui' show Point;
 /// You can use this to manually simulate individual events, but the
 /// simplest way to generate coherent gestures is to use [TestGesture].
 class TestPointer {
-  /// Creates a [TestPointer]. By default, the pointer identifier used is 1, however
-  /// this can be overridden by providing an argument to the constructor.
-  TestPointer([ this.pointer = 1 ]);
+  /// Creates a [TestPointer].
+  ///
+  /// By default, the pointer identifier used is the next available identifier,
+  /// starting with 1. A specific pointer identifier can be explicitly provided
+  /// using the `pointer` argument. Multiple [TestPointer]s created with the
+  /// same pointer identifier will interfere with each other if they are used in
+  /// parallel.
+  factory TestPointer({ int pointer }) {
+    if (pointer == null) {
+      pointer = nextPointer;
+      nextPointer += 1;
+    }
+    return new TestPointer._(pointer);
+  }
+
+  TestPointer._(this.pointer);
+
+  /// The next available pointer identifier.
+  ///
+  /// This is the default pointer identifier that will be used the next time the
+  /// [new TestPointer] constructor is called without an explicit pointer
+  /// identifier.
+  static int nextPointer = 1;
```

Comment:
This creates cross-test dependencies.  Maybe put the counter on the WidgetTester or the test bindings?